### PR TITLE
Merge handlers and middlewares

### PR DESCRIPTION
**Pull request by rossnomann**

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -24,57 +21,28 @@ impl App {
...
-    /// Default values is `ErrorStrategy::Abort`.
-    pub fn middleware_error_strategy(mut self, strategy: ErrorStrategy) -> Self {
-        self.middleware_error_strategy = strategy;
-        self
-    }
-
     /// Set handler error strategy
     ///
     /// See [ErrorStrategy](enum.ErrorStrategy.html) for more information.
     /// Default values is `ErrorStrategy::Abort`.
```
Comment (ark0f) [inline src/app.rs]: Update docs

File: src/app.rs [hunk 2]
```diff
@@ -24,57 +21,28 @@ impl App {
...
 
-    /// Add a regular handler
+    /// Add a handler
     pub fn add_handler(mut self, handler: Handler) -> Self {
         self.handlers.push(handler);
         self
     }
 
     /// Run app
     pub fn run(self, api: Api, method: UpdateMethod) -> impl Future<Item = (), Error = ()> {
```
Comment (ark0f) [inline src/app.rs]: And here

File: src/app.rs [hunk 3]
```diff
@@ -24,57 +21,28 @@ impl App {
     /// Creates a new app
```
Comment (ark0f) [inline src/app.rs]: Docs for `error_strategy` as minimum

File: examples/echo.rs [hunk 4]
```diff
@@ -13,7 +13,7 @@ fn handle_message(context: Context, message: &Message) -> HandlerFuture {
         let api = context.get::<Api>();
         return HandlerFuture::new(api.execute(&method).then(|x| {
             log::info!("sendMessage result: {:?}\n", x);
-            Ok(context)
+            Ok(HandlerResult::Continue(context))
```
Comment (ark0f) [inline examples/echo.rs]: Can we move context outside of handlers?

Comment (rossnomann) [reply examples/echo.rs]: No

**Pull request merged**